modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/portlet/action/MoveKBObjectMVCActionCommand.java

```diff
@@ -52,31 +64,114 @@ protected void doProcessAction(
 			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
-		long resourceClassNameId = ParamUtil.getLong(
-			actionRequest, "resourceClassNameId");
-		long resourcePrimKey = ParamUtil.getLong(
-			actionRequest, "resourcePrimKey");
-		long parentResourcePrimKey = ParamUtil.getLong(
-			actionRequest, "parentResourcePrimKey",
-			KBFolderConstants.DEFAULT_PARENT_FOLDER_ID);
-
-		long kbArticleClassNameId = _portal.getClassNameId(
-			KBArticleConstants.getClassName());
+		boolean dragAndDrop = ParamUtil.getBoolean(
+			actionRequest, "dragAndDrop");
 
-		if (resourceClassNameId == kbArticleClassNameId) {
+		try {
+			long resourceClassNameId = ParamUtil.getLong(
+				actionRequest, "resourceClassNameId");
+			long resourcePrimKey = ParamUtil.getLong(
+				actionRequest, "resourcePrimKey");
 			long parentResourceClassNameId = ParamUtil.getLong(
 				actionRequest, "parentResourceClassNameId",
 				_portal.getClassNameId(KBFolderConstants.getClassName()));
-			double priority = ParamUtil.getDouble(actionRequest, "priority");
+			long parentResourcePrimKey = ParamUtil.getLong(
+				actionRequest, "parentResourcePrimKey",
+				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+			long kbArticleClassNameId = _portal.getClassNameId(
+				KBArticleConstants.getClassName());
+
+			boolean movable = true;
+
+			if (resourceClassNameId == kbArticleClassNameId) {
+				double priority = ParamUtil.getDouble(
+					actionRequest, "priority");
+
+				if (_isDragAndDrop(dragAndDrop)) {
+					KBArticle kbArticle = _kbArticleService.getLatestKBArticle(
+						resourcePrimKey, WorkflowConstants.STATUS_ANY);
+
+					if (kbArticle.getParentResourcePrimKey() ==
+							parentResourcePrimKey) {
+
+						movable = false;
+					}
+
+					priority = kbArticle.getPriority();
+				}
+
+				if (movable) {
+					_kbArticleService.moveKBArticle(
+						resourcePrimKey, parentResourceClassNameId,
+						parentResourcePrimKey, priority);
+				}
```

Comment:
Would it be possible to simplify and reduce the level of nesting? Getting rid of the `movable` variable and complementing the conditions would simplify things quite a bit. For example:
```
if (!_isDragAndDrop(dragAndDrop) {
    _moveKBArticle(...);
}
else {
    KBArticle kbArticle = ...;

    if (kbArticle.getParentResourcePrimKey() != ...) {
        _moveKBArticle(...);
    }
}
```